src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -47,572 +68,581 @@ private static char[] ConvertToLowercaseArray(string text)
+            // we realize that, no matter what, our final edit distance will be more than that threshold
+            // (at which point we can return early).
+            //
+            // Some things are trivially easy to check.  First, the edit distance between two strings is at
+            // *best* the difference of their lengths.  i.e. if i have "aa" and "aaaaa" then the edit
+            // distance is 3 (the difference of 5 and 2).  If our threshold is less then 3 then there
+            // is no way these two strings could match.  So we can leave early if we can tell it would
+            // simply be impossible to get an edit distance within the specified threshold.
+            //
+            // Second, let's look at our matrix again:
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1 
+            //    a |∞ 2
+            //    t |∞ 3
+            //    u |∞ 4
+            //    r |∞ 5
+            //    d |∞ 6
+            //    a |∞ 7
+            //    y |∞ 8           *
+            //
+            // We want to know what the value is at *, and we want to stop as early as possible if it
+            // is greater than our threshold.
+            //
+            // Given the edit distance rules we observe edit distance at any point (i,j) in the matrix will
+            // always be greater than or equal to the value in (i-1, j-1).  i.e. the edit distance of
+            // any two strings is going to be *at best* equal to the edit distance of those two strings
+            // without their final characters.  If their final characters are the same, they'll ahve the
+            // same edit distance.  If they are different, the edit distance will be greater.  Given 
+            // that we know the final edit distance is in the lower right, we can discover something 
+            // useful in the matrix.
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1 
+            //    a |∞ 2
+            //    t |∞ 3 `
+            //    u |∞ 4   `
+            //    r |∞ 5     `
+            //    d |∞ 6       `
+            //    a |∞ 7         `
+            //    y |∞ 8           *
+            //
+            // The slashes are the diagonal leading to the lower right.  The value in the lower right will
+            // be strictly equal to or greater than any value on this diagonal.  Thus, if that value 
+            // exceeds the threshold, we know we can stop immediately as the total edit distance must be
+            // greater than the threshold.
+            //
+            // We can use similar logic to avoid even having to examine more of the matrix when we
+            // have a threshold. First, consider the bottom diagonal.
```

Comment:
"Bottom diagonal" doesn't have meaning yet, as you've only introduced the idea of a single diagonal so far.